pkg/volume/cinder/manager_local.go

```diff
@@ -0,0 +1,559 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cinder
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"go/build"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/volumeactions"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/util/mount"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/cinder/drivers"
+	"k8s.io/kubernetes/pkg/volume/util"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+)
+
+type cdManagerLocal struct {
+	plugin *cinderPlugin
+}
+
+func (cdl *cdManagerLocal) GetName() string {
+	return "local"
+}
+
+// Attaches the volume specified by the given spec to the node with the given Name.
+// On success, returns the device path where the device was attached on the
+// node.
+func (cdl *cdManagerLocal) Attach(spec *volume.Spec, nodeName types.NodeName) (string, error) {
+	volumeSource, readOnly, err := getVolumeSource(spec)
+	if err != nil {
+		return "", err
+	}
+	globalPDPath := makeGlobalPDName(cdl.plugin.host, volumeSource.VolumeID)
+	glog.V(4).Infof("Cinder (%v): Attach %v", cdl.GetName(), volumeSource.VolumeID)
+
+	client, err := newCinderClientFromSecret(volumeSource.SecretRef, cdl.plugin.host)
```

Comment:
Why not call `newCinderClientFromSecret(...)` in `getManager(...)` and stick the resulting `gophercloud.ServiceClient` in the `cdManagerLocal` struct, instead of creating it on every call?